File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    total_words=[]
    def dfs(text):
        if len(text)>0:
            total_words.append(text)

        if len(text)<5:
            for ch in ['A','E','I','O','U']:
                dfs(text+ch)

    dfs('')
    return total_words.index(word)+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
vector<string> words;

void dfs(string cur_word){
    words.push_back(cur_word);
    if (cur_word.length()<5){
        vector<string> tmp_words ={"A","E","I","O","U"};
        for (auto ch:tmp_words)
            dfs(cur_word+ch);
    }
}

int solution(string word) {
    int answer = 0;
    vector tmp_words ={"A","E","I","O","U"};
    for (auto ch:tmp_words)
        dfs(ch);
    
    answer=distance(words.begin(),find(words.begin(),words.end(),word))+1;
    return answer;
}